### DISCUSSION TITLE
Class-scoped generic `using` alias for simplification of generic types

### DISCUSSION THREAD

**Discussion by ero-qt** [dotnet/csharplang#7466] [Language Ideas]
It would be great if it were possible to declare class-scoped `using` alias directives to simplify usage of some generic types within the class;

```cs
class Foo<T, U> { }
class Bar<T>
{
  using Foo<U> = Foo<T, U>;
}
```
This would simplify the use of `Foo` to allow simply use of `Foo<Baz>` instead of `Foo<T, Baz>` within `Bar<T>`.

**Comment by colejohnson66** [comment]
This is a limitation I've questioned. Why is `using` limited to file-scope? I understand function-scope not being permitted as it would interfere with `using` statements, but even that could be worked around; If there's no variable name after the type, it's an alias. Class-scope would at least be nice. I've had a few situations where I redefined constants (with `const <type> <name> = <long-class-name>.<name>`) to simplify typing.

**Comment by HaloFour** [comment]
See: #2981 and #1239

**Comment by CyrusNajmabadi** [comment]
Even if we supported aliases within a type, this would still not work:

> `using Foo<U> = Foo<T, U>;`

That's because aliases do not allow for generic instantiation. That would have to be supported first.
